src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -0,0 +1,219 @@
+                "Move",
+                "AppendAllLines",
+                "AppendAllText",
+                "AppendText",
+                "Delete");
+
+        private readonly ImmutableHashSet<string> SystemReflectionAssemblyMethodMetadataNames = ImmutableHashSet.Create(
+                StringComparer.Ordinal,
+                "GetLoadedModules",
+                "Load",
+                "LoadFile",
+                "LoadFrom",
+                "LoadModule",
+                "LoadWithPartialName",
+                "ReflectionOnlyLoad",
+                "ReflectionOnlyLoadFrom",
+                "UnsafeLoadFrom");
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var compilation = compilationStartAnalysisContext.Compilation;
+                    var attributeTypeSymbols = ImmutableArray.Create(
+                        WellKnownTypes.OnDeserializingAttribute(compilation),
+                        WellKnownTypes.OnDeserializedAttribute(compilation));
+                    var serializableAttributeTypeSymbol = WellKnownTypes.SerializableAttribute(compilation);
+                    var streamingContextTypeSymbol = WellKnownTypes.StreamingContext(compilation);
+
+                    if (serializableAttributeTypeSymbol == null || streamingContextTypeSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var builder = ImmutableArray.CreateBuilder<IMethodSymbol>();
+                    var systemIOFileTypeSymbol = WellKnownTypes.SystemIOFile(compilation);
+
+                    if (systemIOFileTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemIOFileMethodMetadataNames)
+                        {
+                            builder.AddRange(systemIOFileTypeSymbol.GetMembers(targetMethodName).Cast<IMethodSymbol>());
```

Comment:
Just to be more robust, should do a .Where(o => o is IMethodSymbol) before the .Cast().  In case somebody does something weird and references non-standard base class libraries or something.